src/main/webapp/app/pages/userPage/UserPage.tsx

```diff
@@ -126,6 +142,56 @@ export default class UserPage extends React.Component<IUserPage> {
     this.reactions.forEach(disposer => disposer());
   }
 
+  readonly usageDetail = remoteData<Map<string, UsageRecord[]>>({
+    await: () => [],
+    invoke: async () => {
+      this.userUsage = await client.userUsageGetUsingGET({
+        userId: this.user.id.toString(),
+      });
+      const result = new Map<string, UsageRecord[]>();
+      const yearSummary = this.userUsage.summary.year;
+      const yearUsage: UsageRecord[] = [];
+      Object.keys(yearSummary).forEach(resourceEntry => {
+        yearUsage.push({
+          resource: resourceEntry,
+          usage: yearSummary[resourceEntry],
+          time: USAGE_ALL_TIME_VALUE,
+        });
+      });
+      result.set(USAGE_ALL_TIME_KEY, yearUsage);
+
+      const monthSummary = this.userUsage.summary.month;
```

Comment:
When a user has no usage, nor id is invalid, you got not summary data. Here will throw exception causing the promise never been resolved. Thereafter the loading state never changes.